--- a/glibc-2.11/sysdeps/unix/sysv/linux/i386/bits/mman.h	2010-01-26 13:20:00.000000000 +0200
+++ b/glibc-2.11/sysdeps/unix/sysv/linux/i386/bits/mman.h	2010-01-26 13:19:09.000000000 +0200
@@ -92,6 +92,8 @@
 # define MADV_REMOVE	 9	/* Remove these pages and resources.  */
 # define MADV_DONTFORK	 10	/* Do not inherit across fork.  */
 # define MADV_DOFORK	 11	/* Do inherit across fork.  */
+# define MADV_MERGEABLE	 12	/* KSM may merge identical pages.  */
+# define MADV_UNMERGEABLE 13	/* KSM may not merge identical pages.  */
 #endif
 
 /* The POSIX people had to invent similar names for the same things.  */
--- a/glibc-2.11/sysdeps/unix/sysv/linux/x86_64/bits/mman.h	2010-01-26 13:19:52.000000000 +0200
+++ b/glibc-2.11/sysdeps/unix/sysv/linux/x86_64/bits/mman.h	2010-01-26 13:19:20.000000000 +0200
@@ -93,6 +93,8 @@
 # define MADV_REMOVE	 9	/* Remove these pages and resources.  */
 # define MADV_DONTFORK	 10	/* Do not inherit across fork.  */
 # define MADV_DOFORK	 11	/* Do inherit across fork.  */
+# define MADV_MERGEABLE	 12	/* KSM may merge identical pages.  */
+# define MADV_UNMERGEABLE 13	/* KSM may not merge identical pages.  */
 #endif
 
 /* The POSIX people had to invent similar names for the same things.  */
