--- configure.in.orig	2008-08-03
+++ configure.in	2008-08-03
@@ -22,7 +22,7 @@
 
 dnl configure libtool
 AC_DISABLE_STATIC
-AC_LIBLTDL_CONVENIENCE
+AC_WITH_LTDL
 AC_SUBST(LTDLINCL)
 AC_SUBST(LIBLTDL)
 AC_LIBTOOL_DLOPEN
@@ -30,7 +30,6 @@
 AC_PROG_LEX
 AC_PROG_YACC
 PKG_PROG_PKG_CONFIG
-AC_CONFIG_SUBDIRS(libltdl)
 
 AC_MSG_CHECKING([for kernel type ($host_os)])
 case $host_os in
--- Makefile.am.orig	2008-08-03
+++ Makefile.am	2008-08-03
@@ -1,4 +1,4 @@
-SUBDIRS = libltdl src bindings
+SUBDIRS = src bindings
 
 INCLUDES = $(LTDLINCL)
 
