diff -ruN libvirt-0.7.7.orig/src/qemu/qemu_driver.c libvirt-0.7.7/src/qemu/qemu_driver.c
--- libvirt-0.7.7.orig/src/qemu/qemu_driver.c	2010-03-05 16:45:23.000000000 +0100
+++ libvirt-0.7.7/src/qemu/qemu_driver.c	2010-03-26 17:20:50.000000000 +0100
@@ -2360,7 +2360,7 @@
 
         if (hostdev->mode != VIR_DOMAIN_HOSTDEV_MODE_SUBSYS)
             continue;
-        if (hostdev->source.subsys.type != VIR_DOMAIN_HOSTDEV_SUBSYS_TYPE_PCI)
+        if (hostdev->source.subsys.type != VIR_DOMAIN_HOSTDEV_SUBSYS_TYPE_USB)
             continue;
 
         /* Resolve a vendor/product to bus/device */
@@ -2701,6 +2701,11 @@
         return -1;
     }
 
+    /* Must be run before security labelling */
+    DEBUG0("Preparing host devices");
+    if (qemuPrepareHostDevices(driver, vm->def) < 0)
+        goto cleanup;
+
     /* If you are using a SecurityDriver with dynamic labelling,
        then generate a security label for isolation */
     if (driver->securityDriver &&
